packages/babel-register/src/node.js

```diff
@@ -6,7 +6,73 @@ import * as babel from "@babel/core";
 import { OptionManager, DEFAULT_EXTENSIONS } from "@babel/core";
 import { addHook } from "pirates";
 import fs from "fs";
-import path from "path";
+import path, { extname } from "path";
+
+// Technically we could use the "semver" package here, but (for exmaple)
+// parseFloat("4.23.6") returns 4.23 so it's "good enough"
+const BABEL_SUPPORTS_EXTENSIONS_OPTION = parseFloat(babel.version) >= 7.11;
```

Comment:
# :warning: 

- Before merging this PR, I need to add a warning in `new-version` in `Makefile` to replace 7.11 with the actual new version.